app/demo-geant-integration/DetectorConstruction.cc

```diff
@@ -24,6 +24,21 @@
 
 namespace demo_geant
 {
+//---------------------------------------------------------------------------//
+/*!
+ * Set up Celeritas SD options during construction.
+ *
+ * This should be done only during the main/serial thread.
+ */
+DetectorConstruction::DetectorConstruction()
+{
+    auto& sd   = demo_geant::GlobalSetup::Instance()->GetSDSetupOptions();
+    sd.enabled = true;
+    sd.ignore_zero_deposition = true;
```

Comment:
Right now this is a limitation of putting all the different SDs into a single `StepCollector`, for which energy deposition (and indeed, the selection of what properties are saved) is a global option. We should definitely make that more flexible in the future but it would be an optimization (since you can set it to never ignore here and ignore the hits in the SD user code). I'll add a note in the demo code indicating this.